lib/tasks/polls.rake

```diff
@@ -107,4 +107,96 @@ namespace :polls do
       end
     end
   end
+
+  desc "Removes duplicate poll partial results"
+  task remove_duplicate_partial_results: :environment do
+    logger = ApplicationLogger.new
+    duplicate_records_logger = DuplicateRecordsLogger.new
+
+    logger.info "Removing duplicate partial results in polls"
+
+    Tenant.run_on_each do
+      Poll::Question.find_each do |question|
+        manageable_titles = PollPartialResultOptionFinder.new(question).manageable_choices.keys
+
+        question.question_options.each do |option|
+          titles = option.translations.where(title: manageable_titles).select(:title).distinct
+
+          groups = question.partial_results.where(option_id: nil, answer: titles)
+                           .select(:booth_assignment_id, :date)
+                           .group(:booth_assignment_id, :date)
+                           .having("count(*) > 1")
+                           .pluck(:booth_assignment_id, :date)
+
+          groups.each do |booth_assignment_id, date|
+            partial_results = Poll::PartialResult.where(
+              option_id: nil,
+              question_id: question.id,
+              booth_assignment_id: booth_assignment_id,
+              date: date,
+              answer: titles
+            )
+
+            tenant_info = " on tenant #{Tenant.current_schema}" unless Tenant.default?
+
+            amounts_by_id = partial_results.pluck(:id, :amount).to_h
+            if amounts_by_id.values.uniq.size > 1
+              log_message = "Found duplicate partial results with different amounts " \
+                            "for question_id #{question.id}, " \
+                            "booth_assignment_id #{booth_assignment_id} " \
+                            "and date #{date}. " \
+                            "Keeping ID #{partial_results.first.id} " \
+                            "with amount #{partial_results.first.amount}. " \
+                            "Others: #{amounts_by_id.except(partial_results.first.id)}" + tenant_info.to_s
```

Comment:
Maybe, to contrast with "keeping", we can use "deleting" :thinking:. And then clarify what the values in the hash are about.

```suggestion
                            "Deleting partial results with these IDs and amounts: #{amounts_by_id.except(partial_results.first.id)}" + tenant_info.to_s
```

I know the line I wrote is too long; I'll leave fixing that to you :wink:.